rmw_connext_cpp/src/functions.cpp

```diff
@@ -119,13 +134,46 @@ rmw_create_node(const char * name)
     return NULL;
   }
 
-  rmw_node_t * node_handle = new rmw_node_t;
+  rmw_node_t * node_handle = rmw_node_allocate();
+  if (!node_handle) {
+    rmw_set_error_string("failed to allocate memory for node handle");
+    if (dpf_->delete_participant(participant) != DDS_RETCODE_OK) {
+      std::cerr << "leaking participant while handling failue at " <<
+        __FILE__ << ":" << __LINE__;
```

Comment:
We should avoid streaming to stdout / stderr since this will potentially interleave output when being using in multiple threads.